extensions/controllers/sandboxclaim_controller.go

```diff
@@ -48,33 +59,88 @@ type SandboxClaimReconciler struct {
 //+kubebuilder:rbac:groups=extensions.agents.x-k8s.io,resources=sandboxclaims/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=agents.x-k8s.io,resources=sandboxes,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=extensions.agents.x-k8s.io,resources=sandboxtemplates,verbs=get;list;watch
+//+kubebuilder:rbac:groups=policy,resources=poddisruptionbudgets,verbs=get;list;watch;create;update;patch;delete
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 func (r *SandboxClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
 	claim := &extensionsv1alpha1.SandboxClaim{}
 	if err := r.Get(ctx, req.NamespacedName, claim); err != nil {
 		if k8errors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
 		return ctrl.Result{}, fmt.Errorf("failed to get sandbox claim %q: %w", req.NamespacedName, err)
 	}
+	originalClaimStatus := claim.Status.DeepCopy()
+
+	template, err := r.getTemplate(ctx, claim)
+	if err != nil && !k8errors.IsNotFound(err) {
+		// This is a real error, update status and requeue
+		r.computeAndSetStatus(claim, nil, err)
+		// We can't update status if the claim is not found, but we also don't need to.
+		if !k8errors.IsNotFound(err) {
+			if updateErr := r.updateStatus(ctx, originalClaimStatus, claim); updateErr != nil {
+				err = errors.Join(err, updateErr)
+			}
+		}
+		return ctrl.Result{}, err
+	}
+	if k8errors.IsNotFound(err) {
+		// Template not found, set status and stop.
+		r.computeAndSetStatus(claim, nil, ErrTemplateNotFound)
+		if updateErr := r.updateStatus(ctx, originalClaimStatus, claim); updateErr != nil {
+			err = errors.Join(err, updateErr)
+		}
+		return ctrl.Result{}, err
+	}
```

Comment:
Would you refactor the code L77-96 to make it cleaner? There's a lot of dup code, making it hard to read. The only difference is the err you pass to r.computeAndSetStatus. Please combine them.

Reply:
ack